### PR TITLE
Py3 00

### PR DESCRIPTION
**Pull request by pinkavaj**
Hi,

I'm slowly going towards Python 3 port of this library. It would be nice to support both Python 2 and 3 in the same code base. More changes will come latter but it looks like it might work (I have it already done, but needs a lot of clean-up). There is one catch, supporting all of Python 2.6, 2.7 and 3.x might be a bit troublesome. Is there any need to support python 2.6?

### CODE REVIEW

**Comment by mjbrown** [comment]
I will test it on both this weekend.

**Comment by pinkavaj** [comment]
What exactly do you mean by "both"?

**Comment by mjbrown** [comment]
I will test it with 2.7 and 3.5.

**Comment by pinkavaj** [comment]
There are still few patches missing to make it really work with 3.x, this is just try if You are interested in this port. I will try to finish it son and push it. Thanks for fast response.

**Comment by pinkavaj** [comment]
This should do it. I have only tested it on my project (using only api.py). I have only one dongle now, thus I cannot run the test..

Check if you are ok with the changes.

File: bgapi/api.py [hunk 1]
```diff
@@ -201,7 +210,7 @@ def ble_cmd_gap_set_adv_parameters(self, adv_interval_min, adv_interval_max, adv
     def ble_cmd_gap_set_adv_data(self, set_scanrsp, adv_data):
         self.send_command(6, 9, struct.pack('<BB' + str(len(adv_data)) + 's',  set_scanrsp, len(adv_data), adv_data))
     def ble_cmd_gap_set_directed_connectable_mode(self, address, addr_type):
-        self.send_command(6, 10, struct.pack('<6sB', b''.join(chr(i) for i in address), addr_type))
+        self.send_command(6, 10, struct.pack('<6BB', *address, addr_type))
```
Comment (mjbrown) [inline bgapi/api.py]: Python < 3.5 does not allow unnamed arguments after * expression.

File: test/test_client_server.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 from bgapi.module import BlueGigaModule, GATTCharacteristic, GATTService, BlueGigaClient, BlueGigaServer
 from bgapi.cmd_def import gap_discoverable_mode, gap_connectable_mode
+from __future__ import print_function
```
Comment (mjbrown) [inline test/test_client_server.py]: Needs to be the first line in the file.

**Comment by mjbrown** [comment]
Except for the directed connectable problem, the 2-dongle test runs fine in both 2.7 and 3.5.

**Pull request merged**